docs/architecture/psa-thread-safety/psa-thread-safety.md

```diff
@@ -281,40 +281,36 @@ Note that a thread must hold the global mutex when it reads or changes a slot's
 
 #### Slot states
 
-For concurrency purposes, a slot can be in one of three states:
+For concurrency purposes, a slot can be in one of four states:
 
-* UNUSED: no thread is currently accessing the slot. It may be occupied by a volatile key or a cached key.
-* WRITING: a thread has exclusive access to the slot. This can only happen in specific circumstances as detailed below.
-* READING: any thread may read from the slot.
+* EMPTY: no thread is currently accessing the slot, and no information is stored in the slot.
+* FILLING: one thread is currently loading or creating material to fill the slot, this thread is responsible for the next state transition.
+* FULL: the slot contains a key, and any thread is able to use the key after registering as a reader.
```

Comment:
It seems that the thread that created the material is implicitly registered as a reader, and must explicitly unregister_read() when done - is that right? If so, I think we should call that out in the text (this is based on the state diagram, where FILLING goes to FULL+has_readers)

Reply:
The error was in this document, the slot will be left unlocked. I have edited the diagram to fix this.